### PR TITLE
[VIS-849] Feat: Add support for Annotations API

### PR DESCRIPTION
**Pull request by miladcognite**
## Description
* Migrated the code for Annotations API from experimental python SDK, with some minor modifications:
   * Deprecated fields are removed in [37680a9](https://github.com/cognitedata/cognite-sdk-python/pull/928/commits/37680a9e61d25691a0920225ccd8f6cae942d032)
   * Some test fixtures/utils are moved into the test files (see [here](https://github.com/cognitedata/cognite-sdk-python-experimental/blob/cc589b8283a1ae17c5a8c1153d937230a282c731/tests/conftest.py#L11-L49) and [here](https://github.com/cognitedata/cognite-sdk-python-experimental/blob/cc589b8283a1ae17c5a8c1153d937230a282c731/tests/utils.py#L65-L72))
* Upgrade `black` to fix pre-commit issue (https://github.com/psf/black/issues/296)

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change. 
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring. 
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,21 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+
+## [2.50.1] - 2022-06-15
```
Comment (miladcognite) [inline CHANGELOG.md:18]: TODO: update date once PR is reviewed/ready for merge

File: cognite/client/_cognite_client.py [hunk 2]
```diff
@@ -123,6 +124,7 @@ def __init__(
             self._config, api_version="playground", cognite_client=self
         )
         self.transformations = TransformationsAPI(self._config, api_version=self._API_VERSION, cognite_client=self)
+        self.annotations = AnnotationsAPI(self._config, api_version="playground", cognite_client=self)
```
Comment (miladcognite) [inline cognite/client/_cognite_client.py:127]: TODO: update `api_version` once the endpoints have been migrated to V1

**Review by cyriaquebrousse — APPROVED** [review]
Looks good; probably worth mentioning that the code and tests are copied from [the experimental SDK](https://github.com/cognitedata/cognite-sdk-python-experimental/blob/master/cognite/experimental/_api/annotations_v2.py)

File: cognite/client/_version.py [hunk 3]
```diff
@@ -1,2 +1,2 @@
-__version__ = "2.50.0"
+__version__ = "2.50.1"
```
Comment (marvrez) [inline cognite/client/_version.py:1]: Shouldnt we bump the major or minor rather than the patch? See https://semver.org/spec/v2.0.0.html

**Review by marvrez — APPROVED** [review]
LGTM! Added a minor comment regarding the package version

File: cognite/client/_version.py (hunk 3 above)
Comment (miladcognite) [reply cognite/client/_version.py:1]: Nice catch. Looking at previous feature releases (e.g. this [one](https://github.com/cognitedata/cognite-sdk-python/pull/858)https://github.com/cognitedata/cognite-sdk-python/pull/858) it should be minor

Comment (miladcognite) [reply cognite/client/_version.py]: [9f6262b](https://github.com/cognitedata/cognite-sdk-python/pull/928/commits/9f6262b19a5dfd8c454d9811c10b42ebc8f6eeba)

**Review by tuanng-cognite — APPROVED** [review]

**Review by peetcremer1 — APPROVED** [review]
Looks good from my side as well!

File: cognite/client/_api/annotations.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+        if "annotatedResourceIds" in filter:
+            filter["annotatedResourceIds"] = [
+                {to_camel_case(k): v for k, v in f.items()} for f in filter["annotatedResourceIds"]
+            ]
+
+        if "linkedResourceIds" in filter:
+            filter["linkedResourceIds"] = [
+                {to_camel_case(k): v for k, v in f.items()} for f in filter["linkedResourceIds"]
+            ]
```
Comment (peetcremer1) [inline cognite/client/_api/annotations.py:84]: Can be removed as "linkedResourceIds" should never appear in the filter anymore

Comment (miladcognite) [reply cognite/client/_api/annotations.py]: [5cdf073](https://github.com/cognitedata/cognite-sdk-python/pull/928/commits/5cdf073ea643d7be6d7bfbe83874307d072be959)

File: cognite/client/_cognite_client.py (hunk 2 above)
Comment (miladcognite) [reply cognite/client/_cognite_client.py]: [97f72ef](https://github.com/cognitedata/cognite-sdk-python/pull/928/commits/97f72efaf185fe073a6ed3c3e28b1cc72afe8a02)

**Review by ponnapz — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (miladcognite) [reply CHANGELOG.md]: [8372a09](https://github.com/cognitedata/cognite-sdk-python/pull/928/commits/8372a0940ee145a0239706e4e673d1cdb3b0688a)

**Pull request merged**